src/main/java/org/cactoos/collection/CollectionEnvelope.java

```diff
@@ -37,9 +37,15 @@
  * @version $Id$
  * @param <X> Element type
  * @since 0.23
+ * @checkstyle AbstractClassNameCheck (500 lines)
  */
-@SuppressWarnings("PMD.TooManyMethods")
-public class CollectionEnvelope<X> implements Collection<X> {
+@SuppressWarnings(
+    {
+        "PMD.TooManyMethods",
+        "PMD.AbstractNaming"
+    }
+)
+public abstract class CollectionEnvelope<X> implements Collection<X> {
```

Comment:
Rename to **AbstractCollectionEnvelope**

Reply:
@llorllale I'm sorry, but I disagree of this solution. I think the developer/architect/anyone must be free to choose any name he/she want. @yegor256 WDYT about it?

Reply:
@fabriciofx @yegor256 my take: if this is allowed then we should get rid of the `PMD.AbstractNaming` check.